pkg/storage/bucket/client.go

```diff
@@ -39,17 +41,22 @@ const (
 
 	// Filesystem is the value for the filesystem storage backend.
 	Filesystem = "filesystem"
+
+	// validPrefixCharactersRegex allows only alphanumeric characters to prevent subtle bugs and simplify validation
+	validPrefixCharactersRegex = `^[\da-zA-Z]+$`
 )
 
 var (
 	SupportedBackends = []string{S3, GCS, Azure, Swift, Filesystem}
 
-	ErrUnsupportedStorageBackend = errors.New("unsupported storage backend")
+	ErrUnsupportedStorageBackend        = errors.New("unsupported storage backend")
+	ErrInvalidCharactersInStoragePrefix = errors.New("storage_prefix contains invalid characters, it may only contain digits and English alphabet letters")
```

Comment:
Nit: I would suggest to not refer to YAML field in the error message, but say "storage prefix" instead.